Tile doesn't render after zoom

Not sure but think this is a follow up from https://github.com/mapbox/mapbox-gl-native/issues/14523. When testing v7.5.0-alpha.1 of the Android SDK containing https://github.com/mapbox/mapbox-gl-native/pull/14527.

### Before zoom:

> Target: LatLng [latitude=40.723102, longitude=-73.9979, altitude=0.0], Zoom:11.797530889511108, Bearing:14.754321455955505, Tilt:11.962963104248047

![image](https://user-images.githubusercontent.com/2151639/57129224-0cdf6600-6d96-11e9-845f-ee221e32622c.png)

### After zoom:

> Target: LatLng [latitude=40.723102, longitude=-73.9979, altitude=0.0], Zoom:12.32962965965271, Bearing:24.598148703575134, Tilt:19.94444465637207

![image](https://user-images.githubusercontent.com/2151639/57129191-f76a3c00-6d95-11e9-80c6-d09a926cc85f.png)

The tiles in the top aren't loading (the one in the bottom did and is showing more detail). 
fwiw zooming a bit more doesn't load the other tiles. 




cc @pozdnyakov @kkaefer @julianrex

There is a problem with the style - missing sprite image:`[ERROR] {mbgl-glfw}[Style]: Failed to load sprite: HTTP status code 404`. Missing image is preventing from finalizing loading tiles where this image is used |=> tiles are not rendered.

--- a/platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/maps/ImageMissingTest.kt
+++ b/platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/maps/ImageMissingTest.kt
@@ -22,12 +22,19 @@ class ImageMissingTest {
   var rule = ActivityTestRule(EspressoTestActivity::class.java)
 
   private lateinit var mapView: MapView
-  private val latch = CountDownLatch(1)
+  private val latch = CountDownLatch(2)
 
   @Test
   fun testMissingImage() {
     rule.runOnUiThread {
-      initMap().addOnStyleImageMissingListener {
+      initMap(styleJson).addOnStyleImageMissingListener {
+        assertEquals("missing-icon", it)
+        latch.countDown()
+      }
+    }
+
+    rule.runOnUiThread {
+      initMap(styleJsonInvalidSprite).addOnStyleImageMissingListener {
         assertEquals("missing-icon", it)
         latch.countDown()
       }
@@ -38,9 +45,9 @@ class ImageMissingTest {
     }
   }
 
-  private fun initMap(): MapView {
+  private fun initMap(style :String): MapView {
     mapView = rule.activity.findViewById(R.id.mapView)
-    mapView.getMapAsync { it.setStyle(Style.Builder().fromJson(styleJson)) }
+    mapView.getMapAsync { it.setStyle(Style.Builder().fromJson(style)) }
     return mapView
   }
 
@@ -50,6 +57,49 @@ class ImageMissingTest {
       "version": 8,
       "name": "Mapbox Streets",
       "sprite": "mapbox://sprites/mapbox/streets-v8",
+      "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
+      "sources": {
+        "point": {
+          "type": "geojson",
+          "data": {
+            "type": "Feature",
+            "properties": {},
+            "geometry": {
+              "type": "Point",
+              "coordinates": [0, 0]
+            }
+          }
+        }
+      },
+      "layers": [{
+        "id": "bg",
+        "type": "background",
+        "paint": {
+          "background-color": "#f00"
+        }
+      }, {
+        "id": "point",
+        "type": "circle",
+        "source": "point",
+        "paint": {
+          "circle-radius": 100
+        }
+      }, {
+        "id": "icon",
+        "type": "symbol",
+        "source": "point",
+        "layout": {
+          "icon-image": "missing-icon"
+        }
+      }]
+    }
+          """
+
+    private const val styleJsonInvalidSprite = """
+    {
+      "version": 8,
+      "name": "Mapbox Streets",
+      "sprite": "mapbox://sprites/mapbox/invalid",
       "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
       "sources": {
         "point": {

--- a/src/mbgl/style/style_impl.cpp
+++ b/src/mbgl/style/style_impl.cpp
@@ -309,6 +309,9 @@ void Style::Impl::onSpriteError(std::exception_ptr error) {
     lastError = error;
     Log::Error(Event::Style, "Failed to load sprite: %s", util::toString(error).c_str());
     observer->onResourceError(error);
+    // Unblock rendering tiles (even though sprite request has failed).
+    spriteLoaded = true;
+    observer->onUpdate();
 }
 
 void Style::Impl::onLayerChanged(Layer& layer) {

